dax/api.go

```diff
@@ -292,20 +293,50 @@ func (d *Dax) BatchGetItemPagesWithContext(aws.Context, *dynamodb.BatchGetItemIn
 	return d.unImpl()
 }
 
-func (d *Dax) QueryPages(*dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool) error {
-	return d.unImpl()
+func (d *Dax) QueryPages(input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool) error {
+	return d.QueryPagesWithContext(context.Background(), input, fn)
 }
 
-func (d *Dax) QueryPagesWithContext(aws.Context, *dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool, ...request.Option) error {
-	return d.unImpl()
+func (d *Dax) QueryPagesWithContext(ctx aws.Context, input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool, opts ...request.Option) error {
+	for {
+		out, err := d.QueryWithContext(ctx, input, opts...)
+		if err != nil {
+			return err
+		}
+
+		input.ExclusiveStartKey = out.LastEvaluatedKey
```

Comment:
I missed one important one. We need to take a copy of `input` and modify it instead of directly modifying the user supplied parameter.